src/build.rs

```diff
@@ -47,6 +48,42 @@ fn rustc_minor_version() -> Option<u32> {
     otry!(pieces.next()).parse().ok()
 }
 
+/// Checks and returns local and latest versions of wasm-pack
+pub fn check_wasm_pack_versions() -> Result<(String, String), Error> {
+    match wasm_pack_local_version() {
+        Some(local) => {
+            match Crate::return_wasm_pack_latest_version() {
+                Some(latest) => Ok((local, latest)),
+                None => Ok((local, "".to_string()))
+            }
+        },
+        None => bail!("We can't figure out what your wasm-pack version is, make sure the installation path is correct.")
+    }
+}
+
+fn wasm_pack_local_version() -> Option<String> {
+    macro_rules! otry {
+        ($e:expr) => {
+            match $e {
+                Some(e) => e,
+                None => return None,
+            }
+        };
+    }
+
+    let output = otry!(Command::new("wasm-pack").arg("--version").output().ok());
```

Comment:
Sweet!

But instead of spawning a child process to find this out (which has some overhead) I think we’d want to just include the version the binary at compile time.

So maybe use env!(“CARGO_MANIFEST_VERSION”) here

Reply:
Would also be more flexible - for example in the (probably uncommon) case that someone is running a wasm-pack binary that isn’t in their $PATH